lang_java/parsing/parser_java.mly

```diff
@@ -434,6 +434,8 @@ primary_no_new_array:
  | class_literal       { $1 }
  /*(* javaext: ? *)*/
  | method_reference { $1 }
+ /*(* javaext: ? *) */
+ | array_creation_expression_with_initializer { $1 }
```

Comment:
This probably deserves a descriptive comment, especially as it's under the `primary_no_new_array` rule.

Reply:
I agree but I wasn't sure what to do about comments. I don't see many comments in there and I appreciate the ability to see just the structure of the grammar without clutter. I'll describe the problem just before the new rule, I think that's best.

Reply:
added comments and merging.